stdlib/filecmp.pyi

```diff
@@ -13,12 +13,12 @@ __all__ = ["clear_cache", "cmp", "dircmp", "cmpfiles", "DEFAULT_IGNORES"]
 DEFAULT_IGNORES: list[str]
 BUFSIZE: Literal[8192]
 
-def cmp(f1: StrOrBytesPath, f2: StrOrBytesPath, shallow: int | bool = ...) -> bool: ...
+def cmp(f1: StrOrBytesPath, f2: StrOrBytesPath, shallow: bool | Literal[0, 1] = ...) -> bool: ...
 def cmpfiles(
     a: AnyStr | PathLike[AnyStr],
     b: AnyStr | PathLike[AnyStr],
     common: Iterable[AnyStr | PathLike[AnyStr]],
-    shallow: int | bool = ...,
+    shallow: bool | Literal[0, 1] = ...,
```

Comment:
These are cases where we can consider removing the `Literal` at some point. (Otherwise, we'd basically have to replace all instances in argument types with `bool | Literal[0, 1]`.) But no need to do this in this PR.

Reply:
Yes. I held off doing that because I didn't want to introduce any breaking semantic changes in this PR.

Reply:
`bool | int` -> `bool | Literal[0, 1]` is a breaking semantic change. E.g., the following will now no longer type check:
```
def foo(x: int):
    cmp(..., shallow=x)
```
mypy_primer doesn't say anything, so probably nbd, but worth mentioning

Reply:
> `bool | int` -> `bool | Literal[0, 1]` is a breaking semantic change. 

Correct; I should have chosen my words better. I do however feel that it's useful for a user to get an error if, for whatever reason, they try to pass in `3` for this parameter — that probably indicates a logical error in their code somewhere. So my opinion is that this is probably a worthwhile breaking change, and also less breaking than it would have been if we simply changed it to `bool`.

But anyway, as you say, no primer output, so probably nbd :)